File: generated_test_scripts/sprint_1/test_case_1.py

from selenium import webdriver
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time
import sys

# Configurations for headless mode and browser options
options = Options()
options.headless = True
options.add_argument("--disable-notifications")
options.add_argument("--disable-popup-blocking")
options.add_argument("--incognito")

# Initialize the driver
driver = webdriver.Chrome(options=options)
driver.implicitly_wait(5)

try:
    # Step 1: Open the website
    driver.get("https://saucedemo.com/")
    driver.maximize_window()
    time.sleep(5)

    # Step 2: Login
    class LoginPage:
        @staticmethod
        def login(username, password):
            driver.find_element(By.XPATH, '//*[@id="user-name"]').send_keys(username)
            time.sleep(3)
            driver.find_element(By.XPATH, '//*[@id="password"]').send_keys(password)
            time.sleep(3)
            driver.find_element(By.XPATH, '//*[@id="login-button"]').click()
            time.sleep(3)

    LoginPage.login("standard_user", "secret_sauce")

    # Step 3: Add Bike Light and Fleece Jacket to cart
    driver.find_element(By.XPATH, '//*[@id="add-to-cart-sauce-labs-bike-light"]').click()
    time.sleep(3)
    driver.find_element(By.XPATH, '//*[@id="add-to-cart-sauce-labs-bolt-t-shirt"]').click()
    time.sleep(3)

    # Check the cart badge
    cart_count = WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.XPATH, '//*[@id="shopping_cart_container"]/a/span'))
    )
    if cart_count.text != "2":
        raise AssertionError("Cart badge did not display '2' after adding items")

    # Step 4: Reset the cart
    # (Assuming there is a "Reset" or "Remove" option logic here to clear the cart)

    driver.find_element(By.ID, "remove-sauce-labs-bike-light").click()
    time.sleep(3)
    driver.find_element(By.ID, "remove-sauce-labs-bolt-t-shirt").click()
    time.sleep(3)

    # Check cart badge is empty
    try:
        cart_count = WebDriverWait(driver, 10).until(
            EC.invisibility_of_element_located((By.XPATH, '//*[@id="shopping_cart_container"]/a/span'))
        )
    except:
        raise AssertionError("Cart is not empty after reset")

    # Step 5: Add Bolt T-Shirt to cart
    driver.find_element(By.XPATH, '//*[@id="add-to-cart-sauce-labs-bolt-t-shirt"]').click()
    time.sleep(3)

    # Check the cart badge
    cart_count = WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.XPATH, '//*[@id="shopping_cart_container"]/a/span'))
    )
    if cart_count.text != "1":
        raise AssertionError("Cart badge did not display '1' after adding Bolt T-Shirt")

    sys.exit(0)  # Test case passed

except Exception as e:
    print(f"An error occurred: {e}")
    sys.exit(1)  # Test case failed

finally:
    driver.quit()

File: generated_test_scripts/poc_data - sheet1/test_case_1.java

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.util.concurrent.TimeUnit;

public class LoginTest {

    public static void main(String[] args) {
        // Set up Chrome options
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--headless");
        options.addArguments("--incognito");
        options.addArguments("--disable-notifications");
        options.addArguments("--disable-popup-blocking");

        // Initialize WebDriver
        WebDriver driver = new ChromeDriver(options);

        try {
            // Maximize
            driver.manage().window().maximize();

            // Open page
            driver.get("https://practicetestautomation.com/practice-test-login/");
            TimeUnit.SECONDS.sleep(5);

            WebDriverWait wait = new WebDriverWait(driver, 10);

            // Username input
            WebElement usernameField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//input[@name='username']")));
            TimeUnit.SECONDS.sleep(3);
            usernameField.sendKeys("student");

            // Password input
            WebElement passwordField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//input[@name='password']")));
            TimeUnit.SECONDS.sleep(3);
            passwordField.sendKeys("Password123");

            // Submit button
            WebElement submitButton = wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[@type='submit']")));
            TimeUnit.SECONDS.sleep(3);
            submitButton.click();

            // Verification of successful login
            WebElement logoutButton = wait.until(ExpectedConditions.visibilityOfElementLocated(By.linkText("Log out")));
            TimeUnit.SECONDS.sleep(3);
            if (logoutButton.isDisplayed()) {
                System.out.println("Test Passed");
                System.exit(0);
            } else {
                System.out.println("Test Failed");
                System.exit(1);
            }
        } catch (Exception e) {
            e.printStackTrace();
            System.out.println("Test Failed");
            System.exit(1);
        } finally {
            driver.quit();
        }
    }
}
